lib/apivore/rspec_builder.rb

```diff
@@ -14,16 +14,35 @@ module RspecBuilder
 
     @@master_swagger_uri = nil
 
-    def apivore_setup(path, method, response, &block)
-      @@setups[path + method + response] = block
+    # Setup tests against a combination of path, method, and response.
+    # - *keys -> A combination of path, method, and/or response. Blank '' for base setup.
+    # - &block -> Code block to execute to setup the test. A hash of path subsitution parameters can be returned if required.
+    # All matching code blocks are executed, and substitution parameters are merged in order of specificity.
+    def apivore_setup(*keys, &block)
+      @@setups[keys.join] = block
     end
 
     def get_apivore_setup(path, method, response)
-      setup = @@setups[path + method + response]
-      if setup
-        result = instance_eval &setup
+      keys_to_search = [
+        '', # base setup key
+        response,
+        method,
+        path,
+        method + response,
+        # Is there a use-case for the following two? Added for completeness
+        path + response,
+        path + method,
+        path + method + response
```

Comment:
I was thinking of doing something here with `[path, method, response].combination(2)` etc to build the combinations, but I'm not sure that'd be very clear, I thought being explicit about the combinations would be more helpful.

```
a = ['a', 'b', 'c']
['', *a.combination(1).each{}, *a.combination(2).collect{|n| n.join}, a.join]
=> ["", "a", "b", "c", "ab", "ac", "bc", "abc"]
```
Probably doesn't help...